sklearn/model_selection/_validation.py

```diff
@@ -663,13 +668,32 @@ def _score(estimator, X_test, y_test, scorer):
     if isinstance(scorer, dict):
         # will cache method calls if needed. scorer() returns a dict
         scorer = _MultimetricScorer(**scorer)
-    if y_test is None:
-        scores = scorer(estimator, X_test)
-    else:
-        scores = scorer(estimator, X_test, y_test)
 
-    error_msg = ("scoring must return a number, got %s (%s) "
-                 "instead. (scorer=%s)")
+    try:
+        if y_test is None:
+            scores = scorer(estimator, X_test)
+        else:
+            scores = scorer(estimator, X_test, y_test)
+    except Exception:
+        if error_score == 'raise':
+            raise
+        else:
+            if hasattr(scorer, "get_scorers_name"):
+                scores = {
+                    name: error_score for name in scorer.get_scorers_name()
+                }
+            else:
+                scores = error_score
```

Comment:
I think adding a new method in `_MultimetricScorer` almost adds public api.

If we need the keys here, we can do:

```py
if isinstance(scorer, _MultimetricScorer)
	scores = {name: error_score for name in scorer. _scorers}

```

Reply:
yep I add a dilemma when writing these lines :)